operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -124,38 +124,58 @@ func extractGCSConfigSecret(s *corev1.Secret) (*storage.GCSStorageConfig, error)
 	}, nil
 }
 
-func extractS3ConfigSecret(s *corev1.Secret) (*storage.S3StorageConfig, error) {
+func extractS3ConfigSecret(s *corev1.Secret, osEnabled bool) (*storage.S3StorageConfig, error) {
 	// Extract and validate mandatory fields
-	endpoint := s.Data[storage.KeyAWSEndpoint]
-	if len(endpoint) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAWSEndpoint)
-	}
 	buckets := s.Data[storage.KeyAWSBucketNames]
 	if len(buckets) == 0 {
 		return nil, kverrors.New("missing secret field", "field", storage.KeyAWSBucketNames)
 	}
+
+	// Fields related with static authentication
+	endpoint := s.Data[storage.KeyAWSEndpoint]
 	id := s.Data[storage.KeyAWSAccessKeyID]
-	if len(id) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAWSAccessKeyID)
-	}
 	secret := s.Data[storage.KeyAWSAccessKeySecret]
-	if len(secret) == 0 {
-		return nil, kverrors.New("missing secret field", "field", storage.KeyAWSAccessKeySecret)
-	}
-
-	// Extract and validate optional fields
+	// Fields related with STS authentication
+	roleArn := s.Data[storage.KeyAWSRoleArn]
+	var wiTokenFile string
+	audience := s.Data[storage.KeyAWSAudience]
+	// Optional fields
 	region := s.Data[storage.KeyAWSRegion]
 
+	if len(roleArn) == 0 {
+		if len(endpoint) == 0 {
+			return nil, kverrors.New("missing secret field", "field", storage.KeyAWSEndpoint)
+		}
+		if len(id) == 0 {
+			return nil, kverrors.New("missing secret field", "field", storage.KeyAWSAccessKeyID)
+		}
+		if len(secret) == 0 {
+			return nil, kverrors.New("missing secret field", "field", storage.KeyAWSAccessKeySecret)
+		}
+	} else {
+		// In the STS case region is not an optional field
+		if len(region) == 0 {
+			return nil, kverrors.New("missing secret field", "field", storage.KeyAWSRegion)
+		}
+		wiTokenFile = storage.SATokenVolumeK8sDirectory
+		if osEnabled {
+			wiTokenFile = storage.SATokenVolumeOcpDirectory
+		}
```

Comment:
Can we skip doing this extra branching here to improve readibility and making this function OCP-unaware? In fact we have our storage configure method that takes care of patching the envVars anyway. We could make the decision making there? WDYT?